src/components/Tabs.vue

```diff
@@ -37,49 +37,42 @@
 	</div>
 </template>
 
-<script setup>
-// @ts-check
-
+<script setup lang="ts">
+import { TabData } from '../types';
+import { TabsKey } from '../symbols';
 import {
 	computed,
 	provide,
 	reactive,
-	useSlots
+	useSlots,
+	VNode
 } from 'vue';
 
-const props = defineProps( {
-	/**
-	 * Must match the id of one of the tabs. If this prop is not provided, the
-	 * inital active tab will be the first one in the markup structure.
-	 */
-	initialActive: {
-		type: String,
-		required: false
-	}
-} );
+const props = defineProps<{
```

Comment:
I guess if we want to de-duplicate prop typing, going with Typescript over Vue PropTypes makes more sense. I wonder if `vue-docgen-cli` can handle prop definitions/comments in this way; I'll test it out in my VitePress demo.

Reply:
I don't think we *have* to use this feature; pretty sure we can still use regular `defineProps` as well here. I'd be curious if that yields better results.

Reply:
Unfortunately not; the API doesn't work with script setup at all yet, even if you define props the Vue way. But it's good to keep in mind that we could use `defineProps` the Vue way if we decide we prefer that syntax or if it's supported by `vue-docgen-api` earlier than the pure TypeScript version